cmd/postcli/main.go

```diff
@@ -0,0 +1,181 @@
+	flag.BoolVar(&printConfig, "printConfig", false, "print the used config and options")
+	flag.BoolVar(&genProof, "genproof", false, "generate proof as a sanity test, after initialization")
+	flag.StringVar(&opts.DataDir, "datadir", opts.DataDir, "filesystem datadir path")
+	flag.Uint64Var(&opts.MaxFileSize, "maxFileSize", opts.MaxFileSize, "max file size")
+	flag.IntVar(&opts.ComputeProviderID, "provider", opts.ComputeProviderID, "compute provider id (required)")
+	flag.Uint64Var(&cfg.LabelsPerUnit, "labelsPerUnit", cfg.LabelsPerUnit, "the number of labels per unit")
+	flag.BoolVar(&reset, "reset", false, "whether to reset the datadir before starting")
+	flag.StringVar(&idHex, "id", "", "miner's id (public key), in hex (will be auto-generated if not provided)")
+	flag.StringVar(&commitmentAtxIdHex, "commitmentAtxId", "", "commitment atx id, in hex (required)")
+	numUnits := flag.Uint64("numUnits", uint64(opts.NumUnits), "number of units")
+	flag.Parse()
+
+	opts.NumUnits = uint32(*numUnits) // workaround the missing type support for uint32
+}
+
+func processFlags() error {
+	if opts.ComputeProviderID < 0 {
+		return errors.New("-provider flag is required")
+	}
+
+	if commitmentAtxIdHex == "" {
+		return errors.New("-commitmentAtxId flag is required")
+	}
+	var err error
+	commitmentAtxId, err = hex.DecodeString(commitmentAtxIdHex)
+	if err != nil {
+		return fmt.Errorf("invalid commitmentAtxId: %w", err)
+	}
+
+	if idHex == "" {
+		pub, priv, err := ed25519.GenerateKey(nil)
+		if err != nil {
+			return fmt.Errorf("failed to generate identity: %w", err)
+		}
+		id = pub
+		log.Info("cli: generated id: %x", id)
+		saveKey(priv) // The key will need to be loaded in clients for the PoST data to be usable.
+	} else {
+		var err error
+		id, err = hex.DecodeString(idHex)
+		if err != nil {
+			return fmt.Errorf("invalid id: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func main() {
+	parseFlags()
+
+	if printProviders {
+		spew.Dump(gpu.Providers())
+		return
+	}
+
+	if printConfig {
+		spew.Dump(cfg)
+		spew.Dump(opts)
+		return
+	}
+
+	if err := processFlags(); err != nil {
+		log.Panic("cli: %v", err)
+	}
+
+	init, err := initialization.NewInitializer(
+		initialization.WithConfig(cfg),
+		initialization.WithInitOpts(opts),
+		initialization.WithNodeId(id),
+		initialization.WithCommitmentAtxId(commitmentAtxId),
+		initialization.WithLogger(log),
+	)
+	if err != nil {
+		log.Panic(err.Error())
+	}
+
+	if reset {
+		if err := init.Reset(); err != nil {
+			log.Panic("reset error: %v", err)
+		}
+		log.Info("cli: reset completed")
+		return
+	}
+
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+
+	if err := init.Initialize(ctx); err != nil {
+		if err == shared.ErrInitCompleted {
+			log.Panic(err.Error())
+			return
+		}
+		if err == context.Canceled {
+			log.Info("cli: initialization interrupted")
+			return
+		}
+		log.Error("cli: initialization error: %v", err)
+		return
+	}
```

Comment:
`errors.Is` unwraps an error and checks if any (nested) error is of the type, in case the error you are looking for was wrapped:

```suggestion
	err := init.Initialize(ctx)
	switch {
	case errors.Is(err, shared.ErrInitCompleted):
		log.Panic(err.Error())
		return
	case errors.Is(err, context.Canceled):
		log.Info("cli: initialization interrupted")
		return
	case err != nil:
		log.Error("cli: initialization error: %v", err)
		return
	}
```

Reply:
Thanks, committed.